examples/ExactlyOnce/Program.cs

```diff
@@ -15,788 +15,988 @@
+            string transactionalId = null;
+            if (args.Length > 1) transactionalId = args[1];
+
+            IConfiguration configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("./appsettings.json")
+                .Build();
+
+            var adminClientConfig = configuration.GetSection("AdminClient").Get<AdminClientConfig>();
+            using var adminClient = new AdminClientBuilder(adminClientConfig).Build();
+
+            switch (command)
+            {
+                case "create_words":
+                    await new WordCreator().Execute(configuration, transactionalId, adminClient, cts);
+                    break;
+                case "reverse_words":
+                    await new WordReverser().Execute(configuration, transactionalId, adminClient, cts);
+                    break;
+                case "print_words":
+                    await new WordPrinter().Execute(configuration, adminClient, cts);
+                    break;
+                default:
+                    logger.LogError("Unknown command");
+                    break;
+            }
+
+            completedCancellationTokenSource.Cancel();
+        }
+    }
+
+    /// <summary>
+    ///     Error that is thrown if the transactional producer receives a
+    ///     fatal error.
+    /// </summary>
+    public class TransactionalProcessorFatalError : Exception
+    {
+    }
+
+    /// <summary>
+    ///     Represents a single message to be sent
+    ///     to an output topic.
+    /// </summary>
+    public class ProduceMessage<K, V>
     {
         /// <summary>
-        ///     Default timeout used for transaction related operations.
+        ///     Name of the topic to produce to.
         /// </summary>
-        static TimeSpan DefaultTimeout = TimeSpan.FromSeconds(30);
+        public string Topic { get; set; }
 
         /// <summary>
-        ///     The number of partitions we'll use for each topic.
+        ///     Message to produce.
         /// </summary>
-        const int NumPartitions = 12;
+        public Message<K, V> Message { get; set; }
 
         /// <summary>
-        ///     The name of the input Kafka topic containing the lines of text to count the words in.
-        ///         * key = null
-        ///         * value = a line of text (type: string)
+        ///     Delivery handler for this message.
         /// </summary>
-        const string Topic_InputLines = "lines";
+        public Action<DeliveryReport<K, V>> DeliveryHandler { get; set; } = null;
+    }
+
+    /// <summary>
+    ///     The transactional processor allows to use a producer an consumer for transactions,
+    ///     by recreating the producer when a fatal error happens, sending offsets to transaction
+    ///     and handling aborts and retries correctly.
+    /// </summary>
+    public sealed class TransactionalProcessor<KC, VC, KP, VP> : IDisposable
```

Comment:
i think this is getting big enough that it'd be better to split into different files. `Processor` can probably have its own file. maybe keep everything else in this file.